packages/ui/src/components/Visualization/Custom/CustomNode.tsx

```diff
@@ -136,9 +137,17 @@ export const CustomNodeWithSelection: typeof DefaultNode = withSelection()(
     }
 
     if (nodeInteractions.canRemoveStep) {
-      items.push(
-        <ItemDeleteStep key="context-menu-item-delete" data-testid="context-menu-item-delete" vizNode={vizNode} />,
-      );
+      nodeInteractions.canHaveChildren || nodeInteractions.canHaveSpecialChildren
```

Comment:
```suggestion
      const shouldConfirmBeforeDeletion = nodeInteractions.canHaveChildren || nodeInteractions.canHaveSpecialChildren;
```

Then we pass this value to the `ItemDeleteStep` component, so we don't need to create a new one